contrib/tools/workflowcheck/determinism/determinism_test.go

```diff
@@ -0,0 +1,54 @@
+// The MIT License
+//
+// Copyright (c) 2022 Temporal Technologies Inc.  All rights reserved.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package determinism_test
+
+import (
+	"regexp"
+	"testing"
+
+	"go.temporal.io/sdk/contrib/tools/workflowcheck/determinism"
+	"golang.org/x/tools/go/analysis/analysistest"
+)
+
+func Test(t *testing.T) {
+	identRefs := determinism.DefaultIdentRefs.Clone()
+	identRefs["a.BadCall"] = true
+	identRefs["a.BadVar"] = true
+	identRefs["(a.SomeInterface).BadCall"] = true
+	identRefs["a.IgnoredCall"] = false
+	identRefs["os.Stderr"] = false
+	analysistest.Run(
```

Comment:
Can we have a comment here to explain what this is testing?

Reply:
That is documented by the function called: https://pkg.go.dev/golang.org/x/tools/go/analysis/analysistest#Run